File: dummy-repos/python-api-sample/scripts/lint.py
#!/usr/bin/env python3
"""Tiny stdlib-only linter so the dummy repo's `lint` check needs no installs.

Checks every tracked .py file under app/, tests/ and scripts/ for:
  * syntax errors (py_compile)
  * trailing whitespace
  * literal tab indentation
  * lines longer than 120 characters
Exits non-zero (with a report) on the first category of violations found.
"""
from __future__ import annotations

import py_compile
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent
DIRS = ["app", "tests", "scripts"]
MAX_LEN = 120


def main() -> int:
    violations: list[str] = []
    for d in DIRS:
        for path in sorted((ROOT / d).rglob("*.py")):
            try:
                py_compile.compile(str(path), doraise=True)
            except py_compile.PyCompileError as e:
                violations.append(f"{path}: syntax error: {e.msg.strip()}")
                continue
            for n, line in enumerate(path.read_text().splitlines(), 1):
                rel = path.relative_to(ROOT)
                if line.rstrip("\n") != line.rstrip():
                    violations.append(f"{rel}:{n}: trailing whitespace")
                if "\t" in line:
                    violations.append(f"{rel}:{n}: tab character")
                if len(line) > MAX_LEN:
                    violations.append(f"{rel}:{n}: line too long ({len(line)} > {MAX_LEN})")

    if violations:
        print("LINT FAILED:")
        for v in violations:
            print("  " + v)
        return 1
    print("lint ok")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: dummy-repos/node-api-sample/scripts/lint.js
"use strict";

// Dependency-free linter so the `lint` check needs no `npm install`.
// Flags: trailing whitespace, tab indentation, legacy declarations, long lines.

const fs = require("fs");
const path = require("path");

const ROOT = path.resolve(__dirname, "..");
const DIRS = ["src", "scripts", "test"];
const MAX_LEN = 120;
const violations = [];

function walk(dir) {
  if (!fs.existsSync(dir)) return;
  for (const entry of fs.readdirSync(dir, { withFileTypes: true })) {
    const full = path.join(dir, entry.name);
    if (entry.isDirectory()) walk(full);
    else if (entry.name.endsWith(".js")) lintFile(full);
  }
}

function lintFile(file) {
  const rel = path.relative(ROOT, file);
  const lines = fs.readFileSync(file, "utf8").split("\n");
  lines.forEach((line, i) => {
    const n = i + 1;
    if (/\s+$/.test(line)) violations.push(`${rel}:${n}: trailing whitespace`);
    if (/\t/.test(line)) violations.push(`${rel}:${n}: tab character`);
    if (/(^|[^\w.])var\s+[\w$]/.test(line)) {
      violations.push(`${rel}:${n}: use let/const, not legacy var`);
    }
    if (line.length > MAX_LEN) violations.push(`${rel}:${n}: line too long (${line.length} > ${MAX_LEN})`);
  });
}

DIRS.forEach((d) => walk(path.join(ROOT, d)));

if (violations.length) {
  console.error("LINT FAILED:");
  violations.forEach((v) => console.error("  " + v));
  process.exit(1);
}
console.log("lint ok");
